Initialize use cases for series details screen

Ashraf-El-Fallah opened the pull request:
initialize cast , last seasons , recommended series , series details and  top reviews use cases for series details screen

Ashraf-El-Fallah committed:
initialize cast , last seasons , recommended series , series details and  top reviews use cases for series details screen

dev-bilal-azzam commented on domain/series/src/main/java/com/giraffe/series/usecase/GetCastUseCase.kt:
```diff
@@ -0,0 +1,10 @@
+import com.giraffe.series.entity.Cast
+import com.giraffe.series.repository.SeriesRepository
+
+class GetCastUseCase(private val seriesRepository: SeriesRepository) {
```
I think this use case have to be in the person domain not series, Isn't it?

Ashraf-El-Fallah replied:
We include this use case in the series because every series has a cast, so I think it is more dependent on the series.

poula22 replied:
@Bilalazam26 so we need to depend on person domain to get cast?

mohamedebrahem13 committed:
create getPeopleByMovieId method in the repo

mohamedebrahem13 committed:
create getPeopleByMovieId use case

mohamedebrahem13 committed:
add test cases for getPeopleByMovieId use case

dev-bilal-azzam committed:
Merge pull request #170 from Rome-Squad/feature/person-details-domain-and-datasource

Feature/person details domain and DataSource

Mohamed-Ashraf-2210 committed:
refactor app bar component

mohamedebrahem13 committed:
Refactor: Update Person module for movie credits
- Added `CastMovieResponse`, `CrewMovieResponse`, and `CreditsMovieResponse` data classes to handle movie credit information from the remote API.
- Updated `PersonDto` to include `type` (cast/crew) and `movieId` fields.
- Added `getPeopleByMovieId` to `PersonDao` to fetch people associated with a specific movie.
- Modified `PersonResponse` to use `role` instead of `knownForDepartment`.
- Updated `Person` entity to include a `type` field (enum `PersonType` with values `CAST` and `CREW`).
- Updated mapper functions to support the new response types and entity/DTO changes, including mapping `character` to `role` for cast members and `job` to `role` for crew members.

mohamedebrahem13 committed:
feat: Implement fetching people by movie ID

This commit introduces the functionality to fetch people (cast and crew) associated with a specific movie ID.

Key changes:
- Added `getPeopleByMovieId` to `PersonRepository` and its implementation `PersonRepositoryImpl`. This function first checks the local data source and falls back to the remote data source if no local data is found. Fetched remote data is then cached locally.
- Introduced `PersonType` enum (CAST, CREW) to differentiate roles.
- Added `getCreditsByMovieId` to `PersonRemoteDataSource` and its implementation `PersonRemoteDataSourceImp` to fetch credits from the remote API.
- Added `getPeopleByMovieId` to `PersonLocalDataSource` and its implementation `PersonLocalDataSourceImp` to retrieve people from the local database by movie ID.
- Updated `PersonRepositoryImplTest` to reflect changes in the `Person` entity (renamed `knownForDepartment` to `role`).
- Renamed `SearchPeopleByNameUseCaseTest.kt` to `GetPeopleByMovieIdUseCaseTest.kt`.

skaik-mo committed:
Add click handlers to AppBar back button and end icon

skaik-mo committed:
Merge branch 'fix/design-system-component' into components/fix-appbar

skaik-mo committed:
Merge pull request #171 from Rome-Squad/components/fix-appbar

Add click handlers to AppBar back button and end icon

skaik-mo committed:
Refactor: Update AppBar composable parameters and image display

- Renamed `showLogo` parameter to `showImage`.
- Added `image` and `imageSize` parameters to customize the displayed image.
- Updated default values for visibility parameters (`showBackButton`, `showImage`, `showTitle`, `showCaption`, `showEndIcon`) to `false`.
- Added horizontal padding to the AppBar's Row.
- Applied `CircleShape` clipping to the displayed image.

skaik-mo committed:
add `imageSourceToPainter` composable function

This function converts an image source (String URL or Int drawable resource) into a Jetpack Compose Painter. It uses `rememberAsyncImagePainter` for URLs and `painterResource` for drawable IDs.

mohamedebrahem13 committed:
Fix:problem with the get person all times get form remote not local db

Mohamed-Ashraf-2210 committed:
add image mapper

skaik-mo committed:
fix: Add social media click handlers to MainDetails

This commit introduces click handling for social media icons (YouTube, Facebook, Instagram) within the `MainDetails` composable.

- Added `onYoutubeClick`, `onFacebookClick`, and `onInstagramClick` lambda parameters to `MainDetails`.
- Made `SocialMediaComponent` clickable and invoked the `onClick` lambda when clicked.
- Updated string resources for actor birthday and place of birth to dynamically include the respective values.

skaik-mo committed:
fix: GalleryItem, GallerySection, and IconTextBox components

This commit introduces several updates to the GalleryItem, GallerySection, and IconTextBox components.

- `imageSourceToPainter.kt`: Refactored to be an extension function for `Any`.
- `GalleryItem.kt`:
    - Changed the `image` parameter type from `Int?` to `Painter?`.
- `IconTextBox.kt`:
    - Added a `size` modifier to the `Icon` with a value of `16.dp`.
- `GallerySection.kt`:
    - Added an `onShowMoreClick` lambda parameter.
    - Made the "Show More" text clickable.
    - Used `imageSourceToPainter` for gallery items.
    - Adjusted layout weights and modifiers for gallery items.
- `PreviewDetails.kt`:
    - Added `onShowMoreClick` to `GallerySectionPreview`.
- `strings.xml`: Capitalized "Gallery" and "Show More".

Mohamed-Ashraf-2210 committed:
fix some component like (MainInfoCard, MainMovieOrSeriesDetails, MinimizedInfoRow)

skaik-mo committed:
Refactor: SectionTitle and MoviesListSection composable

This commit refactors the `MoviesListSection` composable to utilize the newly introduced `SectionTitle` composable for displaying its title and optional clickable end text.

- The `SectionTitle` composable now handles the layout and styling of the section title and its associated clickable text.
- The `onClickShowMore` parameter in `MoviesListSection` has been renamed to `onClickEndText` for clarity.
- The `clickableText` parameter in `SectionTitle` is now nullable to support cases where no clickable text is needed.

skaik-mo committed:
Merge branch 'fix/design-system-component' into components/fix-appbar

* fix/design-system-component:
  fix some component like (MainInfoCard, MainMovieOrSeriesDetails, MinimizedInfoRow)
  add image mapper

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/utils/imageSourceToPainter.kt

HendSayed25 committed:
delete all hard coded and add string file for ar

HendSayed25 committed:
fix some issues in cards

HendSayed25 committed:
reformat files

HendSayed25 committed:
add needed function to show the date in specific format

HendSayed25 committed:
updated recording to updates in components

HendSayed25 committed:
correct name of string

HendSayed25 committed:
Merge remote-tracking branch 'origin/fix/design-system-component' into fix/design-system-component

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/preview/PreviewDetails.kt

mohamedebrahem13 committed:
create GetPeopleByShowIdUseCase and it's test and add it to di

mohamedebrahem13 committed:
change the name of the response

mohamedebrahem13 committed:
implement remote ds

mohamedebrahem13 committed:
implement person local ds

mohamedebrahem13 committed:
implement mapper and  person repo

mohamedebrahem13 committed:
remove the logs for  person repo

skaik-mo committed:
Merge branch 'fix/design-system-component' into components/fix-appbar

* fix/design-system-component:
  correct name of string
  updated recording to updates in components
  add needed function to show the date in specific format
  reformat files
  fix some issues in cards
  delete all hard coded and add string file for ar

- **ReviewCard:**
    - Updated to use `formatAsMonthDayYear` for date display.
    - Added image source handling using `imageSourceToPainter`.
    - Made reviewerImageSource, reviewerName, and reviewerUsername non-nullable.
- **MainMovieOrSeriesDetails:**
    - Updated poster image source handling.
- **StaffInfoSection:**
    - Changed "Show More" text to be fetched from string resources.
    - Updated divider thickness.
    - Localized title in previews.
- **ReadMoreText:**
    - Updated preview with sample text.
- **Utils:**
    - Added `formatAsMonthDayYear` extension function for LocalDate.
- **SeasonCard:**
    - Added `onClick` parameter.
    - Updated divider thickness.
    - Localized "Rating", "Episodes", and "Year" text.
    - Updated poster image source handling in previews.
- **StarCastSection:**
    - Changed "Show More" text to be fetched from string resources.
- **MainDetails:**
    - Corrected "Instagram" string resource usage.
- **MinimizedInfoRow:**
    - Updated poster image source handling.
- **PreviewDetails:**
    - Updated various component previews with image source handling, localized text, and date formatting.
- **Localization:**
    - Added Arabic (ar-rEG) string resources for details components.
    - Updated English string resources.

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/components/MainDetails.kt
#	presentation/details/src/main/res/values/strings.xml

Mohamed-Ashraf-2210 committed:
refactor component add to collection

Mohamed-Ashraf-2210 committed:
Merge branch 'fix/design-system-component' into components/fix-appbar

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/components/MainDetails.kt
#	presentation/details/src/main/res/values/strings.xml

Mohamed-Ashraf-2210 committed:
Merge remote-tracking branch 'origin/components/fix-appbar' into components/fix-appbar

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/components/MainDetails.kt
#	presentation/details/src/main/res/values/strings.xml

Mohamed-Ashraf-2210 committed:
resolve conflicts

Mohamed-Ashraf-2210 committed:
Merge pull request #173 from Rome-Squad/components/fix-appbar

Refactor AppBar, Sections, and Gallery Components

Ashraf-El-Fallah committed:
remove person from series details entities

skaik-mo committed:
add color animation to rating stars

skaik-mo committed:
Refactor: Standardize modifier parameter order

This commit reorders the parameters in the `MoviesListSection`, `SectionTitle`, and `AppBar` composables to place the `modifier` parameter as the first optional parameter, adhering to common Kotlin coding conventions.

Specifically:
- In `MoviesListSection.kt`, the `modifier` parameter is moved before `endText`.
- In `SectionTitle.kt`, the `modifier` parameter is moved before `clickableText`.
- In `AppBar.kt`, the `enabled = true` parameter, which is the default, is removed from the `clickable` modifier on the back button icon for conciseness.

skaik-mo committed:
Replaced `viewModel` and `single` with the newer `viewModelOf` and `singleOf`

skaik-mo committed:
Refactor: implement animated for MainDetails

- Added `MainDetailsHeader` composable to display a compact header when scrolling.
- Integrated `SharedTransitionLayout` and `AnimatedContent` to animate the transition between the full details view and the header.
- The animation is triggered based on the scroll position.
- Refactored `MainDetails` to support shared element transitions for the actor image and name.
- Adjusted `AppBar` padding to accommodate the new header.
- Updated previews to showcase the animated header functionality.

mohamedebrahem13 committed:
fix the di package use case module

mohamedebrahem13 committed:
fix the test cases after remove the id from the domain entity

mohamedebrahem13 committed:
add room function that insert list of person not one by one

mohamedebrahem13 committed:
gix the repo and the mapper

mohamedebrahem13 committed:
change hte name to Series not show

skaik-mo committed:
Refactor: Make AppBar parameters nullable and remove redundant visibility flags

The `title`, `caption`, `image`, and `endIcon` parameters in the `AppBar` composable are now nullable. This change simplifies the API by removing the need for separate boolean flags (`showTitle`, `showCaption`, `showImage`, `showEndIcon`) to control their visibility.

mohamedebrahem13 committed:
change the name to Series not show

mohamedebrahem13 committed:
Merge pull request #174 from Rome-Squad/feature/person-details-domain-and-datasource

Feature/person details domain and DataSources

mohamedebrahem13 committed:
Merge branch 'develop' into fix/design-system-component

skaik-mo committed:
Merge pull request #175 from Rome-Squad/fix/design-system-component

Fix/design system component

Ashraf-El-Fallah committed:
finish domain and data implement for series details

Ashraf-El-Fallah committed:
Merge remote-tracking branch 'origin/feature/series-details-domain' into feature/series-details-domain

# Conflicts:
#	domain/series/src/main/java/com/giraffe/series/usecase/GetLastSeasonsUseCase.kt
#	domain/series/src/main/java/com/giraffe/series/usecase/GetSeriesDetailsUseCase.kt

Mohamed-Ashraf-2210 committed:
refactor(entities): Clean up SeriesDetails entity and genres handling

- Remove unused `unrequitedVariable` from SeriesDetails entity
- Replace SeriesGenre with `GenreInfo` sealed class to unify:
  - Genre ID (for series search result)
  - Full Genre (for series details result)
- Remove incomplete functions
